sync_check/util/diff.go

```diff
@@ -0,0 +1,577 @@
+package util
+
+import (
+	"bytes"
+	"database/sql"
+	"fmt"
+	"math/rand"
+
+	"github.com/juju/errors"
+	"github.com/ngaut/log"
+)
+
+// Diff contains two sql DB, used for comparing.
+type Diff struct {
+	db1          *sql.DB
+	db2          *sql.DB
+	timeField    string
+	beginTime    string
+	endTime      string
+	splitField   string
+	dbName       string
+	chunkSize    int
+	sample       int
+	checkThCount int
+}
+
+// NewDiff returns a Diff instance.
+func NewDiff(db1, db2 *sql.DB, dbName, timeField, beginTime, endTime, splitField string, chunkSize, sample, checkThCount int) *Diff {
+	return &Diff{
+		db1:          db1,
+		db2:          db2,
+		timeField:    timeField,
+		beginTime:    beginTime,
+		endTime:      endTime,
+		splitField:   splitField,
+		dbName:       dbName,
+		chunkSize:    chunkSize,
+		sample:       sample,
+		checkThCount: checkThCount,
+	}
+}
+
+// Equal tests whether two database have same data and schema.
+func (df *Diff) Equal() (equal bool, err error) {
+	equal = true
+	tbls1, err := getTables(df.db1)
+	if err != nil {
+		err = errors.Trace(err)
+		return
+	}
+
+	tbls2, err := getTables(df.db2)
+	if err != nil {
+		err = errors.Trace(err)
+		return
+	}
+
+	eq := equalStrings(tbls1, tbls2)
+	if !eq {
+		log.Infof("show tables get different table. [source db tables] %v [target db tables] %v", tbls1, tbls2)
+		equal = false
+	}
+
+	for _, tblName := range tbls1 {
```

Comment:
the tbls1 may have many table，need compare all ? u can compare already Synchronized table i guess

Reply:
add a config, user can choose the table to check